packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -1567,3 +1577,83 @@ class TextStyle with Diagnosticable {
     styles.add(EnumProperty<TextOverflow>('${prefix}overflow', overflow, defaultValue: null));
   }
 }
+
+/// Interpolate between two lists of [FontVariation] objects.
+///
+/// Variations are paired by axis, and interpolated using [FontVariation.lerp].
+///
+/// This algorithm is O(N), but the constant factor varies based on the input.
+///
+/// It is fastest when the lists contain the same axes ([FontVariation.axis]) in
+/// the same order. The result is again in the same order, and no attempt is
+/// made to detect or remove duplicates in this process.
+///
+/// When the order of items in the two input lists vary, the constant factor
+/// increases substantially, as it involves creating two maps and a set,
+/// inserting every variation in both lists into the maps and the set, and then
+/// iterating over them to recreate the list.
+///
+/// In this case, the resulting order is arbitrary. Duplicates are dropped; in
+/// each list, the last [FontVariation] for any particular axis is the one used
+/// to compute the value for that axis. Values that only appear on one side are
```

Comment:
Does it make sense to have a default value for each axis then? Like the weight defaults to 400 if one side is null? Or the developer should always make sure the axis pairs match if the lerp function is used for an animated transition?

Reply:
Fonts have a default value, but we have no way to find out what it is (because, for one, we don't know what font is going to be used, and it could be a different font for each character in the string, due to font fallbacks).

If the developer wants to animate something, they'll specify both ends, in practice.